middleware/throttle.go

```diff
@@ -34,21 +54,46 @@ type token struct{}
 
 // throttler limits number of currently processed requests at a time.
 type throttle struct {
```

Comment:
Since it's a convention to use "-er" type names in Go (writer, reader, encoder, scanner, replacer etc.), I'd prefer throttler.